src/Microsoft.Health.Fhir.Core/Features/Operations/Import/ImportProcessingTask.cs

```diff
@@ -0,0 +1,274 @@
+                    }
+                    catch (Exception ex)
+                    {
+                        // ignore exception for progresss update
+                        _logger.LogInformation(ex, "Failed to update context.");
+                    }
+                }
+
+                // Pop up exception during load & import
+                // Put import task before load task for resource channel full and blocking issue.
+                try
+                {
+                    await importTask;
+                }
+                catch (TaskCanceledException)
+                {
+                    throw;
+                }
+                catch (OperationCanceledException)
+                {
+                    throw;
+                }
+                catch (Exception ex)
+                {
+                    _logger.LogError(ex, "Failed to import data.");
+                    throw new RetriableTaskException("Failed to import data.", ex);
+                }
+
+                try
+                {
+                    await loadTask;
+                }
+                catch (TaskCanceledException)
+                {
+                    throw;
+                }
+                catch (OperationCanceledException)
+                {
+                    throw;
+                }
+                catch (Exception ex)
+                {
+                    _logger.LogError(ex, "Failed to load data.");
+                    throw new RetriableTaskException("Failed to load data", ex);
+                }
+
+                return new TaskResultData(TaskResult.Success, JsonConvert.SerializeObject(result));
+            }
+            catch (TaskCanceledException canceledEx)
+            {
+                _logger.LogInformation(canceledEx, "Data processing task is canceled.");
+                return new TaskResultData(TaskResult.Canceled, JsonConvert.SerializeObject(result));
+            }
+            catch (OperationCanceledException canceledEx)
+            {
+                _logger.LogInformation(canceledEx, "Data processing task is canceled.");
+                return new TaskResultData(TaskResult.Canceled, JsonConvert.SerializeObject(result));
+            }
+            catch (RetriableTaskException)
+            {
+                throw;
+            }
+            catch (Exception ex)
+            {
+                _logger.LogInformation(ex, "Critical error in data processing task.");
+
+                throw new RetriableTaskException(ex.Message);
+            }
+            finally
+            {
+                if (!_cancellationTokenSource.IsCancellationRequested)
+                {
+                    _cancellationTokenSource.Cancel();
+                }
+            }
+        }
+
+        public void Cancel()
+        {
+            if (!_cancellationTokenSource.IsCancellationRequested)
```

Comment:
same~ !_cancellationTokenSource?.IsCancellationRequested

Reply:
yes, we should check this.